### PR TITLE
Re-enable formatting in the `lsp`

### LINKED ISSUE #2802: Re-enable the formatting capabilities of the `lsp`
This includes the removal of `SwayConfig` being replaced by the formatter's `Config` and converting the `serde_json::Value` to its corresponding `ConfigOptions`.

### PR DESCRIPTION
**Pull request by eureka-cpu**
Opening for visibility

Closes #2802

### CODE REVIEW

File: sway-lsp/src/core/session.rs [hunk 1]
```diff
@@ -127,7 +129,10 @@ impl Session {
     // update sway config
     pub fn update_config(&self, options: Value) {
```
Comment (JoshuaBatty) [inline sway-lsp/src/core/session.rs:130]: We can remove this function entirely as we aren't allowing users to update the config from the plugin anymore.

File: sway-lsp/src/core/session.rs [hunk 2]
```diff
@@ -350,8 +355,8 @@ impl Session {
         if let Some(document) = self.documents.get(url.path()) {
             match self.config.read() {
                 std::sync::LockResult::Ok(config) => {
-                    let config: SwayConfig = *config;
-                    let mut formatter = Formatter::from(config);
+                    let config: Config = *config;
```
Comment (JoshuaBatty) [inline sway-lsp/src/core/session.rs:358]: let's just call `swayfmt::Config::default()` here directly.

File: sway-lsp/src/core/session.rs [hunk 3]
```diff
@@ -38,7 +40,7 @@ pub struct CompiledProgram {
 #[derive(Debug)]
 pub struct Session {
     pub documents: Documents,
-    pub config: RwLock<SwayConfig>,
+    pub config: RwLock<Config>,
```
Comment (JoshuaBatty) [inline sway-lsp/src/core/session.rs:43]: Let's remove config from the Session struct and just initialize swayfmt where it's used.

**Review by kayagokalp — APPROVED** [review]
utACK

**Review by JoshuaBatty — APPROVED** [review]
Nice one, thanks!

**Pull request merged**